combustion-android-ble/src/main/java/inc/combustion/framework/ble/DeviceStatus.kt

```diff
@@ -70,13 +73,22 @@ internal data class DeviceStatus(
             val probeID = if(modeColorId != null) ProbeID.fromUByte(modeColorId) else ProbeID.ID1
             val probeMode = if(modeColorId != null) ProbeMode.fromUByte(modeColorId) else ProbeMode.Normal
 
+            // use status if available
+            val status = if (data.size > 22)
+                data.sliceArray(STATUS_RANGE)[0]
+            else
+                null
+
+            val batteryStatus = if(status != null) ProbeBatteryStatus.fromUByte(status) else ProbeBatteryStatus.Ok
```

Comment:
Just a note, you can use the [Elvis operator](https://kotlin-quick-reference.com/156-R-elvis-operator.html) to turn this line into a statement instead of a conditional.  

I probably should have made the same comment above on `probeID` and `probeMode`.  I'm easy on the approach, just want to point out.

Reply:
It'd be nice to be consistent--I'm happy to make the changes to `probeID` and `probeMode`. What do you think?